### PR TITLE
Add doxygen doc in nav2_bt_navigator header files

### PR DESCRIPTION
**Pull request by shivaang12**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1256 |

---

## Description of contribution in a few bullet points

---

## Future work that may be required in bullet points
* Need feedback!

### CODE REVIEW

File: nav2_bt_navigator/include/nav2_bt_navigator/bt_navigator.hpp [hunk 1]
```diff
@@ -30,34 +30,82 @@
 
 namespace nav2_bt_navigator
 {
-
+/**
+ * @class nav2_bt_navigator::BtNavigator
```
Comment (SteveMacenski) [inline nav2_bt_navigator/include/nav2_bt_navigator/bt_navigator.hpp]: I'd add a description about it being an action server

Comment (shivaang12) [reply nav2_bt_navigator/include/nav2_bt_navigator/bt_navigator.hpp]: How about this?
/**
 * @class nav2_bt_navigator::BtNavigator
 * @brief An action server that uses behavior tree for navigating a robot to its
 *  goal position.
 */

Comment (SteveMacenski) [reply nav2_bt_navigator/include/nav2_bt_navigator/bt_navigator.hpp]: LGTM

**Comment by SteveMacenski** [comment]
Ignore CI, other stuff that isnt docs related is broken, but do look at the test builds to make sure you linted correctly

**Comment by shivaang12** [comment]
Cool!

**Comment by shivaang12** [comment]
I reviewed the failed tests, and linters are good!

**Comment by SteveMacenski** [comment]
Just waiting on second approver.

Feel free to do `nav2_planner` next ;) then we'll have all the main action servers documented

**Comment by shivaang12** [comment]
Already on it!

**Pull request merged**